README.md

```diff
@@ -1,7 +1,334 @@
+
+```csharp
+string sourceId = "your Push API source ID";
+PushDocument document = new PushDocument("https://coveo.com") {
+    ClickableUri = "https://www.coveo.com",
+    ModifiedDate = DateTime.UtcNow
+};
+
+document.AddMetadata("title", "Coveo's home page.");
+
+PushDocumentHelper.SetContent(document, "this is Coveo's website home page.");
+
+client.DocumentManager.AddOrUpdateDocument(sourceId, document, null);
+```
+
+**Good to know:**
+* The `SetContent` method has an overload taking a Stream instead of a string.
+* The PushDocumentHelper class has also the method `SetContentFromFile` taking a file path as an argument.
+* The third argument in `AddOrUpdateDocument` is the ordering ID. If you don't provide a value, the SDK will create one using a timestamp to ensure the changes are performed in the order they were received. For more information about ordering ID, visit [Understanding the orderingId Parameter](https://docs.coveo.com/en/147/cloud-v2-developers/understanding-the-orderingid-parameter).
+* The call returns the generated ordering ID if you did not specify one. You can store it. It can be useful to delete a batch of documents.
+
+### Pushing a document with big properties
+In case the size of the document you just created is too big, the SDK does not handle it if you use the call to index a single document. If the whole document is larger than 256MB, it will throw an `ArgumentException`. If the document is larger than 5MB, it will throw a `CoveoPlatformException` including the status code (500) and `Internal server error` inside the `ErrorMessage` property. Take note that if you always use the batched call the SDK will handle it automatically.
+In that case you have 2 options:
+* You can add the document using the batched call `client.DocumentManager.AddOrUpdateDocuments`
+* You can also check for the document size and then decide if you use `client.DocumentManager.AddOrUpdateDocument` or `client.DocumentManager.AddOrUpdateDocuments`
+```csharp
+document.JsonObjectSize > Constants.COMPRESSED_DATA_MAX_SIZE_IN_BYTES
+```
+
+### Pushing a document with a large binary data
+The Push API has a hard limit of 5MB (compressed) for the document's binary data. When the data exceeds that value we need to request an upload URI to an S3 bucket to put the document's binary data in it and refer to that ID when pushing the document in the Push API. However, lucky you, the SDK handles this automatically. Thus, the SDK compresses the content, and if it exceeds 5MB (compressed), it will do the needed logic. Take note that the maximum size of a document is 256MB (compressed). 
+
+### Delete a single document
+```csharp
+string documentId = "https://coveo.com";
+client.DocumentManager.DeleteDocument(sourceId, documentId, null);
+```
+**Good to know:**
+* Prefer deleting documents in batches to save API calls and to have a better application performance.
```

Comment:
"Prefer deleting" -> "Favor deleting"
("prefer" is about taste rather than recommending)

Reply:
```suggestion
* Favor deleting documents in batches to save API calls and to have a better application performance.
```